### DISCUSSION TITLE
Is it possible to ignore files that would be copied in `cibuidlwheel` docker container ?

### DISCUSSION THREAD

**Discussion by FirelightFlagboy** [pypa/cibuildwheel#1495] [Q&A]
Hello, I'm working on a project where we use `cibuildwheel` to build our wheel.

I'm facing a problem where `cibuildwheel` copies files that I don't want to be copied (like `node_modules` files that aren't used by the python project).
At best, that slow down the process because it copies unnecessary files (`node_modules` folder are heavy :stuck_out_tongue:)
At worst, It failed to build the wheel because it have copied some configuration files that are specific for my setup and would not work on the container.

It is possible to configure `cibuildwheel` to not copy some specified files (like `.gitignore` or `.dockerignore`) ?

**Comment by mara004** [comment]
That's a great question.

We have the same problem with pypdfium2, which can have an in-tree checkout of its underlying C++ library, pdfium. This can be huge, so cibuildwheel gets stuck at the `Copying project into container` step.
For local testing, this means we first have to set up a fresh checkout and sync it with the working copy, which is a bit of a nuisance.

As an aside, I'm wondering why cibuildwheel doesn't provide an option to mount the project directory into the container (i.e. shared with the host) instead of copying it? At least that's what we're doing in a custom docker test step.
